### PR TITLE
Multiple bug fixes. 

### PR DESCRIPTION
**Pull request by Kilowhisky**
Fixes issues
https://github.com/kurko/ember-localstorage-adapter/issues/123
https://github.com/kurko/ember-localstorage-adapter/issues/68

### CODE REVIEW

**Comment by kurko** [comment]
Can you fix these failures as well, @Kilowhisky?

**Comment by Kilowhisky** [comment]
@kurko  the build failures you're seeing is because of this bug.  https://github.com/kurko/ember-localstorage-adapter/issues/119 Nothing to do with the changes i've made.

I can give it a shot but i'm not very well versed in the test-suite.

**Comment by Kilowhisky** [comment]
Actually the errors are because of breaking changes in  ember data >= 1.13.0.  

In order to fix them you either need to specify the ember data dependency at '1.0.0-beta.19.2' or declare possible version requirement going forward.

**Comment by Kilowhisky** [comment]
Also is there a reason why the recursive model loading only travels 1 level down?

**Comment by kurko** [comment]
I'm a little wary of merging this without certainty that the tests are actually passing for this set of features... I'm not using the lib right now in any project, so it's hard for me to work on it.

**Comment by Kilowhisky** [comment]
So i have a branch that almost passes the tests but is failing on one last bug that i think is in ember data but others don't seem believe so or are staying silent. Can you take a look at this test and give me your thoughts on how to proceed? 

https://github.com/Kilowhisky/ember-localstorage-adapter/tree/ember-1-13

Here is my open bug report on ember-data btw. 

https://github.com/emberjs/data/issues/3614

**Comment by kurko** [comment]
For some reason, it's extremely hard to follow the https://github.com/kurko/ember-localstorage-adapter/compare/master...Kilowhisky:ember-1-13 diff. It looks like everything is being replaced.

I think that as for the ED bug, you'd need to open a PR there with a failing test in ED (not local storage). I've been on this road for almost 3 years and I now they will not give too much attention unless you're talking about ED. Anything on other adapters seem to simply be "the problem is the adapter".

**Comment by kurko** [comment]
Regarding

```
{
  'list': {
    records: {
      'l1': { id: 'l1', name: 'one', b: true, items: ['i1', 'i2'] },
      'l2': { id: 'l2', name: 'two', b: false, items: [] },
      'l3': { id: 'l3', name: 'three', b: false, items: [] }
    }
  }
}
```

I think we need to filter this because you really shouldn't return items there (list `l1`). If you set `async: true`, then it's fine because a second call will be made to get `items`.

But if you set `async: false`, then you should be returning the included data altogether in the same response.

**Comment by Kilowhisky** [comment]
The diff is because of a line ending screw up. I'll try and see if i can revert it.

**Comment by Kilowhisky** [comment]
Here is a branch with a much cleaner diff.
https://github.com/Kilowhisky/ember-localstorage-adapter/tree/ember-1.13.+

**Comment by kurko** [comment]
Right. Can you reopen against the new branch? :smile:
